pkg/planner/core/tiflash_selection_late_materialization.go

```diff
@@ -218,6 +218,7 @@ func removeSpecificExprsFromSelection(physicalSelection *PhysicalSelection, expr
 	conditions := physicalSelection.Conditions
 	for i := len(conditions) - 1; i >= 0; i-- {
 		if expression.Contains(physicalSelection.SCtx().GetExprCtx().GetEvalCtx(), exprs, conditions[i]) {
+			// TODO: Investigate why we cannot replace this with slices.Delete(conditions, i, i+1)
```

Comment:
@mjonss please follow up. Maybe try using `slices.DeleteFunc()` instead?

Reply:
Running `TestFTSSyntax` fails on this. A child plan is using the same slice and will crash, since after delete the length will be the same for the copy, but the end element will be `nil`. Solved by first cloning the slice, and then DeleteFunc and using the result for `physicalSelection.Conditions`.
@Rustin170506 is that common that the same slice is copied in multiple places, so similar bugs may appear in other places?